buildkite/scripts/cache-through.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -eou pipefail
+set +x
+
+if [[ $# -ne 2 ]]; then
+  echo "Usage: $0 <path-to-file> <postprocess-run-preprocess-in-docker-cmd>"
+  exit 1
+fi
+
+# download gsutil if it doesn't exist
+# TODO: Bake this into the agents
+if [[ ! -f ./google-cloud-sdk/bin/gsutil ]]; then
+  echo "Downloading gsutil because it doesn't exist"
+  wget https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-296.0.1-linux-x86_64.tar.gz && tar -zxf google-cloud-sdk-296.0.1-linux-x86_64.tar.gz
+
+  tar -zxf google-cloud-sdk-296.0.1-linux-x86_64.tar.gz
+
+  echo "$BUILDKITE_GS_APPLICATION_CREDENTIALS_JSON" > /tmp/gcp_creds.json
+
+  export GOOGLE_APPLICATION_CREDENTIALS=/tmp/gcp_creds.json && ./google-cloud-sdk/bin/gcloud auth activate-service-account bk-large@o1labs-192920.iam.gserviceaccount.com --key-file /tmp/gcp_creds.json
```

Comment:
haha yikes

Reply:
I think we'll always do it on "large" since the opam step is heavy anyway (at least until we introduce other stages -- hopefully we'll have this baked into the agent by then)

Reply:
Sure, makes sense and the service account should be changed to a more general setting (e.g. bk-east) once *coda-automation* [#386](https://github.com/CodaProtocol/coda-automation/pull/386) in merged.